tests/Integration/Behaviour/Features/Context/Domain/CustomerGroupFeatureContext.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Gherkin\Node\TableNode;
+use Exception;
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Command\AddCustomerGroupCommand;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Query\GetCustomerGroupForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\QueryResult\EditableCustomerGroup;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\ValueObject\GroupId;
+use PrestaShop\PrestaShop\Core\Domain\Product\QueryResult\ProductForEditing;
+use Tests\Integration\Behaviour\Features\Context\CommonFeatureContext;
+use Tests\Integration\Behaviour\Features\Context\Domain\Product\UpdateTagsFeatureContext;
+use Tests\Integration\Behaviour\Features\Context\Util\DataComparator;
+
+class CustomerGroupFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * "When" steps perform actions, and some of them store the latest result
+     * in this variable so that "Then" action can check its content
+     *
+     * @var mixed
+     */
+    protected $latestResult;
+
+    /**
+     * Registry to keep track of created/edited customer groups using references
+     *
+     * @var int[]
+     */
+    protected $registry = [];
+
+    /**
+     * @When /^I create a Customer Group "(.+)" with the following details:$/
+     *
+     * @param string $customerGroupReference
+     * @param TableNode $table
+     *
+     * @throws Exception
+     */
+    public function createCustomerUsingCommand(string $customerGroupReference, TableNode $table)
+    {
+        $data = $this->localizeByRows($table);
+        $commandBus = $this->getCommandBus();
+
+        $command = new AddCustomerGroupCommand(
+            $data['name'],
+            $data['reduction'],
+            $data['priceDisplayMethod'],
+            $data['showPrice']
+        );
+
+        /** @var GroupId $id */
+        $id = $commandBus->handle($command);
+        $this->latestResult = $id->getValue();
+        $this->registry[$customerGroupReference] = $id->getValue();
```

Comment:
there is a SharedStorage already for that, use that to save references. And get rid of $latestResult & $registry, im pretty sure you will be able to make it work without them.

Reply:
We don't want our context to be stateful, so the usage of local private fields to store values is prohibited. And yet we do need to store some data so they can be reused in other steps later, especially all the mapping between references and their associated IDs

As @zuk3975 suggested we use the shared storage for this:
```suggestion
        $this->getSharedStorage()->set($customerGroupReference, $id->getValue());
```

Reply:
Done